src/lxc/conf.c

```diff
@@ -1650,110 +1647,57 @@ static int lxc_setup_ttydir_console(const struct lxc_rootfs *rootfs,
+	if (ret < 0 || (size_t)ret >= sizeof(path))
 		return -1;
 
-	/* When we are asked to setup a console we remove any previous
-	 * /dev/console bind-mounts.
-	 */
-	if (console->path && !strcmp(console->path, "none")) {
-		struct stat st;
-		ret = stat(path, &st);
-		if (ret < 0) {
-			if (errno == ENOENT)
-				return 0;
-			SYSERROR("failed stat() \"%s\"", path);
-			return -errno;
-		}
-
-		/* /dev/console must be character device with major number 5 and
-		 * minor number 1. If not, give benefit of the doubt and assume
-		 * the user has mounted something else right there on purpose.
-		 */
-		if (((st.st_mode & S_IFMT) != S_IFCHR) || major(st.st_rdev) != 5 || minor(st.st_rdev) != 1)
-			return 0;
-
-		/* In case the user requested a bind-mount for /dev/console and
-		 * requests a ttydir we move the mount to the
-		 * /dev/<ttydir/console.
-		 * Note, we only move the uppermost mount and clear all other
-		 * mounts underneath for safety.
-		 * If it is a character device created via mknod() we simply
-		 * rename it.
-		 */
-		ret = safe_mount(path, lxcpath, "none", MS_MOVE, NULL, rootfs->mount);
-		if (ret < 0) {
-			if (errno != EINVAL) {
-				ERROR("failed to MS_MOVE \"%s\" to \"%s\": %s", path, lxcpath, strerror(errno));
-				return -errno;
-			}
-			/* path was not a mountpoint */
-			ret = rename(path, lxcpath);
-			if (ret < 0) {
-				ERROR("failed to rename \"%s\" to \"%s\": %s", path, lxcpath, strerror(errno));
-				return -errno;
-			}
-			DEBUG("renamed \"%s\" to \"%s\"", path, lxcpath);
-		} else {
-			DEBUG("moved mount \"%s\" to \"%s\"", path, lxcpath);
-		}
-
-		/* Clear all remaining bind-mounts. */
+	if (file_exists(path)) {
 		ret = lxc_unstack_mountpoint(path, false);
 		if (ret < 0) {
 			ERROR("failed to unmount \"%s\": %s", path, strerror(errno));
 			return -ret;
 		} else {
 			DEBUG("cleared all (%d) mounts from \"%s\"", ret, path);
 		}
-	} else {
-		if (file_exists(path)) {
-			ret = lxc_unstack_mountpoint(path, false);
-			if (ret < 0) {
-				ERROR("failed to unmount \"%s\": %s", path, strerror(errno));
-				return -ret;
-			} else {
-				DEBUG("cleared all (%d) mounts from \"%s\"", ret, path);
-			}
-		}
+	}
 
-		if (safe_mount(console->name, lxcpath, "none", MS_BIND, 0, rootfs->mount) < 0) {
-			ERROR("failed to mount '%s' on '%s'", console->name, lxcpath);
-			return -1;
+	fd = open(path, O_CREAT | O_EXCL, S_IXUSR | S_IXGRP | S_IXOTH);
+	if (fd < 0) {
+		if (errno != EEXIST) {
+			SYSERROR("failed to create console");
+			return -errno;
 		}
-		DEBUG("mounted \"%s\" onto \"%s\"", console->name, lxcpath);
+	} else {
+		close(fd);
 	}
 
-	/* create symlink from rootfs /dev/console to '<ttydir>/console' */
-	ret = snprintf(lxcpath, sizeof(lxcpath), "%s/console", ttydir);
-	if (ret < 0 || (size_t)ret >= sizeof(lxcpath))
-		return -1;
-
-	ret = unlink(path);
-	if (ret && errno != ENOENT) {
-		SYSERROR("error unlinking %s", path);
+	if (chmod(console->name, S_IXUSR | S_IXGRP | S_IXOTH)) {
```

Comment:
That's probably a line from me but could you please do:
```
ret = chmod(console->name, S_IXUSR | S_IXGRP | S_IXOTH);
if (ret < 0) {
        /* handle error */
}
```

Reply:
Fixed